File: 45-jump-game-ii/jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        int end = 0;
        int jump = 0;
        int maxm = 0;
        if(n == 1){
            return 0;
        }
        for(int i=0; i<n - 1; i++){
            maxm = max(maxm, i + nums[i]);
            if( i == end){
                jump++;
                end = maxm;
            }
        }
        return jump;
    }
};

File: 45-jump-game-ii/jump-game-ii.py
class Solution:
    def jump(self, nums: list[int]) -> int:
        cnt = 0
        end = 0
        maxjump = 0 
        n = len(nums)
        for i in range(n - 1):
            maxjump = max(maxjump, i + nums[i])
            if i == end:
                cnt += 1
                end = maxjump
        return cnt
        

        